README.md

```diff
@@ -3,48 +3,200 @@
+the
+[Protobuf wire format](https://developers.google.com/protocol-buffers/docs/encoding).
 It is inspired by, and is significantly based on,
-[DER ASCII](https://github.com/google/der-ascii), a similar tool for working with
-DER and BER, wire formats of ASN.1.
+[DER ASCII](https://github.com/google/der-ascii), a similar tool for working
+with DER and BER, wire formats of ASN.1.
 
-Unlike most Protobuf tools, it is completely ignorant of schemata specified in `.proto`
-files; it has just enough knowledge of the wire format to provide primitives for 
-constructing messages (such as field tags, varints, and length prefixes). A disassembler
-is included that uses heuristics to try convert encoded Protobuf into Protoscope,
-although the heuristics are necessarily imperfect.
+Unlike most Protobuf tools, it is normally ignorant of schemata specified in
+`.proto` files; it has just enough knowledge of the wire format to provide
+primitives for constructing messages (such as field tags, varints, and length
+prefixes). A disassembler is included that uses heuristics to try convert
+encoded Protobuf into Protoscope, although the heuristics are necessarily
+imperfect.
 
-We provide the Go package `github.com/protocolbuffers/protoscope`, as well as the `protoscope`
-tool, which can be installed with the Go tool via
+We provide the Go package `github.com/protocolbuffers/protoscope`, as well as
+the `protoscope` tool, which can be installed with the Go tool via
 
-    go install github.com/protocolbuffers/protoscope/cmd/protoscope...@latest
+```
+go install github.com/protocolbuffers/protoscope/cmd/protoscope...@latest
+```
 
-These tools may be used to create test inputs by taking an existing proto,
-dissassembling with `protoscope`, making edits, and then reassembling with
-`protoscope -s`. This avoids having to manually fix up all the length prefixes.
-They may also be used to inspect proto files (or things that look like them.)
+For the language specification and basic examples, see
+[language.txt](/language.txt). Example disassembly can be found under
+[./testdata](/testdata).
 
-For the language specification and basic examples, see [language.txt](/language.txt).
-Example disassembly can be found under [./testdata](/testdata).
+## Cookbook
 
-## Backwards compatibility
+Protoscope can be used in a number of different ways to inspect or create binary
+Protobuf data. This isn't the full breadth of usecases, but they are the ones
+Protoscope (and its ancestor, DER ASCII) were designed for.
+
+### Exploring Binary Dumps
+
+Sometimes, while working on a library that emits wire format, it may be
+necessary to debug the precise output of a test failure. If your test prints out
+a hex string, you can use the `xxd` command to turn it into raw binary data and
+pipe it into `protoscope`:
+
+```sh
+$ cat hexdata.txt
+0a400a26747970652e676f6f676c65617069732e636f6d2f70726f746f332e546573744d65737361676512161005420e65787065637465645f76616c756500000000
```

Comment:
I'd explain what this is ...which I think is an encoded Any?

Reply:
Friendly ping!

Reply:
Clarified.